language/move-prover/stackless-bytecode-generator/tests/borrow/basic_test.exp

```diff
@@ -0,0 +1,277 @@
+
+fun TestBorrow::test1(): TestBorrow::R {
+    var r: TestBorrow::R
+    var r_ref: &mut TestBorrow::R
+    var x_ref: &mut u64
+    var $t3: u64
+    var $t4: TestBorrow::R
+    var $t5: &mut TestBorrow::R
+    var $t6: &mut TestBorrow::R
+    var $t7: &mut u64
+    var $t8: u64
+    var $t9: &mut u64
+    var $t10: TestBorrow::R
+    $t3 := 3
+    $t4 := pack TestBorrow::R($t3)
+    r := $t4
+    $t5 := borrow_local(r)
+    r_ref := $t5
+    $t6 := move(r_ref)
+    $t7 := borrow_field<TestBorrow::R>.x($t6)
+    x_ref := $t7
+    $t8 := 0
+    $t9 := move(x_ref)
+    write_ref($t9, $t8)
+    $t10 := move(r)
+    return $t10
+}
+
+
+fun TestBorrow::test2(x_ref: &mut u64, v: u64) {
+    var $t2: u64
+    var $t3: &mut u64
+    $t2 := copy(v)
+    $t3 := move(x_ref)
+    write_ref($t3, $t2)
+    return ()
+}
+
+
+fun TestBorrow::test3(r_ref: &mut TestBorrow::R, v: u64) {
+    var x_ref: &mut u64
+    var $t3: &mut TestBorrow::R
+    var $t4: &mut u64
+    var $t5: &mut u64
+    var $t6: u64
+    $t3 := move(r_ref)
+    $t4 := borrow_field<TestBorrow::R>.x($t3)
+    x_ref := $t4
+    $t5 := move(x_ref)
+    $t6 := copy(v)
+    TestBorrow::test2($t5, $t6)
+    return ()
+}
+
+
+fun TestBorrow::test4(): TestBorrow::R {
+    var r: TestBorrow::R
+    var r_ref: &mut TestBorrow::R
+    var $t2: u64
+    var $t3: TestBorrow::R
+    var $t4: &mut TestBorrow::R
+    var $t5: &mut TestBorrow::R
+    var $t6: u64
+    var $t7: TestBorrow::R
+    $t2 := 3
+    $t3 := pack TestBorrow::R($t2)
+    r := $t3
+    $t4 := borrow_local(r)
+    r_ref := $t4
+    $t5 := move(r_ref)
+    $t6 := 0
+    TestBorrow::test3($t5, $t6)
+    $t7 := move(r)
+    return $t7
+}
+
+
+fun TestBorrow::test5(r_ref: &mut TestBorrow::R): &mut u64 {
+    var $t1: &mut TestBorrow::R
+    var $t2: &mut u64
+    $t1 := move(r_ref)
+    $t2 := borrow_field<TestBorrow::R>.x($t1)
+    return $t2
+}
+
+
+fun TestBorrow::test6(): TestBorrow::R {
+    var r: TestBorrow::R
+    var r_ref: &mut TestBorrow::R
+    var x_ref: &mut u64
+    var $t3: u64
+    var $t4: TestBorrow::R
+    var $t5: &mut TestBorrow::R
+    var $t6: &mut TestBorrow::R
+    var $t7: &mut u64
+    var $t8: &mut u64
+    var $t9: u64
+    var $t10: TestBorrow::R
+    $t3 := 3
+    $t4 := pack TestBorrow::R($t3)
+    r := $t4
+    $t5 := borrow_local(r)
+    r_ref := $t5
+    $t6 := move(r_ref)
+    $t7 := TestBorrow::test5($t6)
+    x_ref := $t7
+    $t8 := move(x_ref)
+    $t9 := 0
+    TestBorrow::test2($t8, $t9)
+    $t10 := move(r)
+    return $t10
+}
+
+============ after pipeline `borrow` ================
+
+fun TestBorrow::test1(): TestBorrow::R {
+    var r: TestBorrow::R
+    var r_ref: &mut TestBorrow::R
+    var x_ref: &mut u64
+    var $t3: u64
+    var $t4: TestBorrow::R
+    var $t5: &mut TestBorrow::R
+    var $t6: &mut TestBorrow::R
+    var $t7: &mut u64
+    var $t8: u64
+    var $t9: &mut u64
+    var $t10: TestBorrow::R
+    // live_refs:  borrowed_by:  borrows_from:
```

Comment:
Perhaps we can suppress this if all empty.

Reply:
Done.